segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/WriteOperation.java

```diff
@@ -400,13 +418,14 @@ private void addSystemLogRecord(List<SystemJournal.SystemJournalRecord> systemLo
      * Write to chunk.
      */
     private CompletableFuture<Void> writeToChunk(MetadataTransaction txn,
-                                                    SegmentMetadata segmentMetadata,
-                                                    InputStream data,
-                                                    ChunkHandle chunkHandle,
-                                                    ChunkMetadata chunkWrittenMetadata,
-                                                    long offsetToWriteAt,
-                                                    int bytesCount) {
-        Preconditions.checkState(0 != bytesCount, "Attempt to write zero bytes. Segment=%s Chunk=%s offsetToWriteAt=%s", segmentMetadata, chunkWrittenMetadata, offsetToWriteAt);
+                                                 SegmentMetadata segmentMetadata,
+                                                 InputStream data,
+                                                 ChunkHandle chunkHandle,
+                                                 ChunkMetadata chunkWrittenMetadata,
+                                                 long offsetToWriteAt,
+                                                 int bytesCount,
+                                                 AtomicReference<byte[]> expectedContent) {
```

Comment:
Good point. All we need is that variable needs to be either final/effectively final or volatile to avoid thread visibility issues. 

Fixed. Made it final instead of making it volatile via AtomicReference.